src/daemon/daemon.cpp

```diff
@@ -57,6 +57,8 @@ Daemon* Daemon::instance() {
 bool Daemon::activate(const InterfaceConfig& config) {
   Q_ASSERT(wgutils() != nullptr);
 
+  m_state = Activating;
```

Comment:
Is it worth checking `m_state` first? I assume we only ever go from inactive->activating so other states could hint at an underlying issue

```c++
if (m_state != Inactive) {
    return false;
}
```

Reply:
It's possible this method will be called when active in the case of server switching.